homeassistant/components/broadlink/remote.py

```diff
@@ -336,10 +350,74 @@ async def _async_learn_command(self, command):
                 await asyncio.sleep(1)
                 try:
                     code = await self._device.async_request(self._device.api.check_data)
-
                 except (ReadError, StorageError):
                     continue
+                return b64encode(code).decode("utf8")
+            raise TimeoutError("No code received")
```

Comment:
You could give the user more information about this error.
```suggestion
            raise TimeoutError(f"No ir code received within {LEARNING_TIMEOUT.seconds} seconds")
```